### PR TITLE
Add support for Log Analytics on logging-bucket module and  bump provider version

### PR DESCRIPTION
**Pull request by lcaggio**
In this PR I add support for Log Analytics on logging-bucket module.

Issue https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/issues/1422

### CODE REVIEW

File: modules/logging-bucket/README.md [hunk 1]
```diff
@@ -20,6 +20,23 @@ module "bucket" {
...
+  parent      = var.project_id
+  id          = "mybucket"
+  log_analytics = {
+    enable = true
+    linked_dataset = {
+      link_id = "log"
+    }
+  }
+}
+# tftest modules=1 resources=2 inventory=log_analytics.yaml
```
Comment (juliocc) [inline modules/logging-bucket/README.md:38]: Can you extend the example to include the creation of the dataset with the bigquery module?

File: modules/logging-bucket/variables.tf [hunk 2]
```diff
@@ -37,6 +37,16 @@ variable "location" {
   default     = "global"
 }
 
+variable "log_analytics" {
```
[the remark this answers is not in the document]
Comment (lcaggio) [reply modules/logging-bucket/variables.tf:40]: Log analytics and linked datasets are 2 separated features. You can enable Log Analytics without making them available in BQ as dataset. I see the use case to not enable linked dataset, I would keep it as 2 separate variables.

What's your view?

File: modules/logging-bucket/README.md (hunk 1 above)
Comment (lcaggio) [reply modules/logging-bucket/README.md:38]: The dataset is created automatically by the resource. No need to create id in advance, actually it fails if you pass an existing dataset.

File: modules/logging-bucket/variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (juliocc) [reply modules/logging-bucket/variables.tf:40]: Your approach makes sense. I thought enabling log analytics required the linked dataset

**Review by juliocc — APPROVED** [review]

**Pull request merged**